er: new release

CamStan opened the pull request:
er now has a new release, v0.0.4

This adds that new version to the package, updates the url, and updates the hash of v0.0.3 for the new url.

This also updates the KVTree dependency as MPI is required to be enabled in KVTree for er to work.
rankstr is now also required by er for recently added tests.

CamStan committed:
er: new release

er now has a new release, v0.0.4

This adds that new version to the package, updates the url, and
updates the hash of v0.0.3 for the new url.

This also updates the KVTree dependency as MPI is required to be
enabled in KVTree for er to work.
rankstr is now also required by er for recently added tests.

alalazo commented on var/spack/repos/builtin/packages/er/package.py:
```diff
@@ -29,6 +31,7 @@ def cmake_args(self):
         if self.spec.satisfies('platform=cray'):
             args.append("-DER_LINK_STATIC=ON")
         args.append("-DWITH_KVTREE_PREFIX=%s" % self.spec['kvtree'].prefix)
+        args.append("-DWITH_RANKSTR_PREFIX=%s" % self.spec['rankstr'].prefix)
```
I think this line will break for earlier versions, since `rankstr` is not a dependency in that case.

CamStan replied:
Ah, yup. Stuck this in a conditional.

CamStan committed:
add version constraint when appending rankstr arg

This fixes the cmake args to only append the rankstr prefix when
using the latest version or newer.

Also defines spec at beginning of args in order to shorten line
lengths and avoid line-too-long checks

CamStan commented:
Pinging @alalazo, this is ready for another review.
